rules-tests/DeadCode/Rector/If_/RemoveUnusedNonEmptyArrayBeforeForeachRector/Fixture/skip_property_dim_fetch.php.inc

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Rector\Tests\DeadCode\Rector\If_\RemoveUnusedNonEmptyArrayBeforeForeachRector\Fixture;
+
+class SkipPropertyDimFetch
+{
+    public const DEFAULT_GROUP = 'default';
+
+    /**
+     * @var array<string, list<self>>
+     */
+    private static $groups = [];
+
+    public static function getFootnotes($group = self::DEFAULT_GROUP): array
+    {
+        if (! empty(self::$groups[$group])) {
```

Comment:
since [we already skip "regular" dim fetch](https://github.com/staabm/rector-src/blob/dim-fetch/rules-tests/DeadCode/Rector/If_/RemoveUnusedNonEmptyArrayBeforeForeachRector/Fixture/skip_dim_fetch.php.inc), I think it makes sense to also skip it for this example

Reply:
is it possible to to check native ArrayDimFetch type for this case?

```php
if ($foreachExpr instanceof Expr\ArrayDimFetch) {
   $nativeType = $this->nodeTypeResolver-getNativeType(...
}
```

Reply:
Do you have a test case in mind which fails atm?

Reply:
something like assign in prev stmt:

```php
self::$groups[$group] = ["foo"];

if (! empty(self::$groups[$group]))

Reply:
fixed with https://github.com/rectorphp/rector-src/pull/5166/commits/0edaf2da087f9ea3c01ad95f89d01e01ea702e48

Reply:
could you fix phpstan notice? thank you.

Reply:
good catch :-). fixed.